File: fxgl-samples/src/main/java/sandbox/tiled/mario/FSMState.java
/*
 * FXGL - JavaFX Game Library. The MIT License (MIT).
 * Copyright (c) AlmasB (almaslvl@gmail.com).
 * See LICENSE for details.
 */

package sandbox.tiled.mario;

import com.almasb.fxgl.entity.Entity;

/**
 * TODO: interface?
 * on enter / exit?
 * @author Almas Baimagambetov (almaslvl@gmail.com)
 */
public abstract class FSMState {

    protected abstract void onUpdate(Entity entity, double tpf);
}

File: fxgl/src/main/kotlin/com/almasb/fxgl/ai/goap/FSMState.kt
/*
 * FXGL - JavaFX Game Library. The MIT License (MIT).
 * Copyright (c) AlmasB (almaslvl@gmail.com).
 * See LICENSE for details.
 */

package com.almasb.fxgl.ai.goap

import com.almasb.fxgl.entity.Entity

/**
 * Adapted from https://github.com/sploreg/goap
 * Original source: C#, author: Brent Anthony Owens.
 *
 * @author Almas Baimagambetov (almaslvl@gmail.com)
 */
interface FSMState {

    fun update(fsm: FSM, entity: Entity)
}